src/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnosticsForSpan.cs

```diff
@@ -170,21 +170,20 @@ public async Task GetAsync(ArrayBuilder<DiagnosticData> list, CancellationToken
                         {
                             if (includeSyntax)
                             {
-                                syntaxAnalyzers.Add(stateSet);
+                                syntaxAnalyzers.Add(analyzer);
                             }
 
                             if (includeSemantic)
                             {
-                                var stateSets = GetSemanticAnalysisSelectedStates(
-                                    stateSet.Analyzer, _incrementalAnalysis,
+                                var selectedAnalyzers = GetSemanticAnalysisSelectedAnalyzers(
```

Comment:
> var selectedAnalyzers = GetSemanticAnalysisSelectedAnalyzers( [](http://example.com/codeflow?start=31&length=62)

personal preference, this would be more clear to me inlined

Reply:
7ef54be935a